src/errors.rs

```diff
@@ -14,35 +23,194 @@ pub fn handle_panic(info: &PanicInfo<'_>, send_app_instructions: &SyncSender<App
         None => info.payload().downcast_ref::<String>().map(|s| &**s),
     };
 
+    let err_ctx: ErrorContext = OPENCALLS.with(|ctx| ctx.borrow().clone());
+
     let backtrace = match (info.location(), msg) {
         (Some(location), Some(msg)) => format!(
-            "\nthread '{}' panicked at '{}': {}:{}\n{:?}",
+            "{:#?}\n\u{1b}[0;0mError: \u{1b}[0;31mthread '{}' panicked at '{}': {}:{}\n\u{1b}[0;0m{:?}",
+            err_ctx,
             thread,
             msg,
             location.file(),
             location.line(),
             backtrace
         ),
         (Some(location), None) => format!(
-            "\nthread '{}' panicked: {}:{}\n{:?}",
+            "{:#?}\n\u{1b}[0;0mError: \u{1b}[0;31mthread '{}' panicked: {}:{}\n\u{1b}[0;0m{:?}",
+            err_ctx,
             thread,
             location.file(),
             location.line(),
             backtrace
         ),
         (None, Some(msg)) => format!(
-            "\nthread '{}' panicked at '{}'\n{:?}",
-            thread, msg, backtrace
+            "{:#?}\n\u{1b}[0;0mError: \u{1b}[0;31mthread '{}' panicked at '{}'\n\u{1b}[0;0m{:?}",
+            err_ctx, thread, msg, backtrace
+        ),
+        (None, None) => format!(
+            "{:#?}\n\u{1b}[0;0mError: \u{1b}[0;31mthread '{}' panicked\n\u{1b}[0;0m{:?}",
+            err_ctx, thread, backtrace
         ),
-        (None, None) => format!("\nthread '{}' panicked\n{:?}", thread, backtrace),
     };
 
     if thread == "main" {
         println!("{}", backtrace);
         process::exit(1);
     } else {
         send_app_instructions
-            .send(AppInstruction::Error(backtrace))
+            .send((AppInstruction::Error(backtrace), err_ctx))
             .unwrap();
     }
 }
+
+#[derive(Clone)]
+pub struct ErrorContext {
+    calls: ArrayVec<[ContextType; MAX_THREAD_CALL_STACK]>,
+}
+
+impl ErrorContext {
+    pub fn new() -> Self {
+        Self {
+            calls: ArrayVec::new(),
+        }
+    }
+
+    pub fn add_call(&mut self, call: ContextType) {
+        self.calls.push(call);
+        OPENCALLS.with(|ctx| *ctx.borrow_mut() = self.clone());
+    }
+}
+
+impl Debug for ErrorContext {
```

Comment:
Just curious, why `Debug` and not `Display`?

Reply:
No particular reason. I just never used `Display` before :sweat_smile: . I can switch to it if you want.

Reply:
Both work, but IMO in this case `Display` is more correct.